core/lib/rom/types.rb

```diff
@@ -22,5 +25,30 @@ def ForeignKey(relation, type = Types::Int)
         type.meta(foreign_key: true, target: relation)
       end
     end
+
+    module CoercibleMethods
+      def JsonRead(symbol_keys = false, type = Types::Hash)
+        Types.Constructor(type) do |value|
+          next Hash[value] if value.respond_to?(:to_hash)
```

Comment:
I don't think we need this check in ROM 4.0, `value` is expected to string and we don't process it twice. I hope 😬 However, in 3.0 it's required for rom-repo